sdn_tests/testplans/SUBSCRIBE-Modes-Levels-Test-Plan.md

```diff
@@ -0,0 +1,2150 @@
+# Overview
+
+This document aims to outline the approach for testing the functionality of different SUBSCRIBE modes of the gNMI protocol as part of GPINs OpenConfig end-to-end testing. Subscriptions in gNMI can be periodic or event driven. Modifiers can also be applied to request heartbeat updates or filter out unchanged values.
+
+This document also covers testing for different subscription levels. The covered subscription levels are root, subtree and leaf. They will be tested for each subscription mode.
+
+# Background
+
+With SONiC as the network operating system (NOS) for GPINS, gNMI is responsible for monitoring, streaming telemetry, and configuration management.  Broadcom's Unified Management Framework (UMF) provides gNMI streaming telemetry based on the standard OpenConfig model.
```

Comment:
gNMI is responsible for monitoring, streaming telemetry, and configuration management. Sonic-Gnmi streaming telemetry is based on the standard OpenConfig model.